privaterelay/tests/mgmt_update_phone_remaining_stats_tests.py

```diff
@@ -26,69 +31,256 @@
+    assert relay_number.remaining_seconds == settings.MAX_MINUTES_PER_BILLING_CYCLE * 60
+
+
+def test_phone_subscriber_w_phones_reset_1_day_ago_does_not_update_stats(
+    patch_datetime_now, phone_user
+):
+    expected_now = patch_datetime_now
+    profile = Profile.objects.get(user=phone_user)
+    reset_datetime = expected_now - timedelta(1)
+    profile.date_phone_subscription_reset = reset_datetime
+    profile.save()
+    baker.make(RealPhone, user=phone_user, verified=True)
+    relay_number = baker.make(
+        RelayNumber, user=phone_user, remaining_texts=10, remaining_seconds=15
     )
-    assert len(updated_profiles) == 1
-    assert profile in updated_profiles
 
+    num_profiles_w_phones, num_profiles_updated = update_phone_remaining_stats()
 
-def test_one_account_with_phones_checked_31_day_ago_with_relay_number(phone_user):
+    profile.refresh_from_db()
+    assert profile.date_phone_subscription_reset == reset_datetime
+    assert num_profiles_w_phones == 1
+    assert num_profiles_updated == 0
+    relay_number.refresh_from_db()
+    assert relay_number.remaining_texts == 10
+    assert relay_number.remaining_seconds == 15
+
+
+def test_phone_subscriber_wo_date_phone_subscription_reset_and_no_relay_number_reset_date_updated(
+    patch_datetime_now, phone_user
+):
+    expected_now = patch_datetime_now
+    profile = Profile.objects.get(user=phone_user)
+    profile.save()
+
+    num_profiles_w_phones, num_profiles_updated = update_phone_remaining_stats()
+
+    profile.refresh_from_db()
+    assert profile.date_phone_subscription_reset == expected_now
+    assert num_profiles_w_phones == 1
+    assert num_profiles_updated == 1
+
+
+def test_phone_subscriber_w_date_phone_subscription_reset_31_days_ago_and_no_relay_number_reset_date_updated(
+    patch_datetime_now, phone_user
+):
+    expected_now = patch_datetime_now
+    profile = Profile.objects.get(user=phone_user)
+    profile.date_phone_subscription_reset = expected_now - timedelta(31)
+    profile.save()
+
+    num_profiles_w_phones, num_profiles_updated = update_phone_remaining_stats()
+
+    profile.refresh_from_db()
+    assert profile.date_phone_subscription_reset == expected_now
+    assert num_profiles_w_phones == 1
+    assert num_profiles_updated == 1
+
+
+def test_phone_subscriber_with_phones_reset_31_day_ago_phone_limits_updated(
+    patch_datetime_now, phone_user
+):
+    expected_now = patch_datetime_now
     profile = Profile.objects.get(user=phone_user)
-    pre_update_datetime = datetime.now(timezone.utc) - timedelta(31)
-    profile.date_phone_subscription_checked = pre_update_datetime
+    profile.date_phone_subscription_reset = expected_now - timedelta(31)
+    profile.save()
+    _make_real_phone(
+        phone_user, verified=True
+    )  # update other tests to use this util func
+    relay_number = _make_relay_number(phone_user)
+    relay_number.remaining_texts = 10
+    relay_number.remaining_seconds = 15
+    relay_number.save()
```

Comment:
suggestion (blocking): the other tests do this using `baker`. should this test also use `baker`?

Reply:
Resolved in 1fe0140

Reply:
Resolved in 46bcf43